third_party/xla_client/pjrt_computation_client.cc

```diff
@@ -524,13 +536,31 @@ PjRtComputationClient::ExecuteReplicated(
   // Required as of cl/518733871
   execute_options.use_major_to_minor_data_layout_for_callbacks = true;
 
+  std::optional<std::vector<PjRtFuture<Status>>> returned_futures(devices.size());
```

Comment:
Technically, we should add this to the ExecuteOptions arg, instead of creating this inside the execution function -- and then we could do check if that option field is present before taking the action. The async future should outlive the main process calling the function here, so functionally this should be good, I think. FYI.